crates/uv/src/commands/project/init.rs

```diff
@@ -322,3 +346,79 @@ async fn init_project(
 
     Ok(())
 }
+
+/// Initializes the correct VCS system based on the provided config.
+fn init_vcs(path: &Path, vcs: VersionControl) -> Result<()> {
+    match vcs {
+        VersionControl::Git => {
+            if path.join(".git").try_exists()? {
+                warn_user!(
+                    "Git repository already exists at `{}`",
+                    path.simplified_display()
+                );
+            } else {
+                init_git_repo(path)?;
+            }
+
+            // Create the `.gitignore` if it does not already exist.
+            let gitignore = path.join(".gitignore");
+            if !gitignore.try_exists()? {
+                fs_err::write(
+                    gitignore,
+                    indoc::indoc! {r"
+                        # Python generated files
+                        __pycache__/
+                        *.py[oc]
+                        build/
+                        dist/
+                        wheels/
+                        *.egg-info
+
+                        # venv
+                        .venv
+                    "},
+                )?;
+            }
+        }
+        VersionControl::None => {}
+    };
+
+    Ok(())
+}
+
+/// Check if the path is inside a VCS repository.
+///
+/// Currently only supports Git.
+fn existing_vcs_repo(dir: &Path) -> bool {
+    is_inside_git_work_tree(dir)
+}
+
+/// Check if the path is inside a Git work tree.
+fn is_inside_git_work_tree(dir: &Path) -> bool {
+    Command::new("git")
+        .arg("rev-parse")
+        .arg("--is-inside-work-tree")
+        .current_dir(dir)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .status()
+        .map(|status| status.success())
+        .unwrap_or(false)
+}
+
+/// Run `git init` in the given directory.
+fn init_git_repo(dir: &Path) -> Result<()> {
+    if !Command::new("git")
```

Comment:
Might be worth using `which` (similar to interpreter search) to find `git` first and use its full bin path, and that also allows us to only run this when git is found in the path

Reply:
👍 agreed